### PR TITLE
Prevent .deb packages from being opened with archive manager.

### PR DESCRIPTION
**Pull request by Ruk33**
## Fixes

Issue Number: https://github.com/lbryio/lbry-desktop/issues/7495

## What is the current behavior?

With Ubuntu >= 20 .deb packages are opened with archive explorer.

## What is the new behavior?

`.deb` packages are properly installed and the app relaunches when the installation finishes.

image

image

image

^ if there is an error (ie, incorrect password or permission wasn't granted), we let the user know.

image

When the installation finishes, the app will relaunch automatically.

### How to test
From `/lbry-desktop/ui/redux/actions/app.js`, replace lines 218:224 with:

```js
      // if (
      //   upgradeAvailable &&
      //   !selectModal(state) &&
      //   (!selectIsUpgradeSkipped(state) || remoteVersion !== selectRemoteVersion(state))
      // ) {
      //   dispatch(doOpenModal(MODALS.UPGRADE));
      // }
      dispatch(doOpenModal(MODALS.UPGRADE));
```

And update line 213 with:

```js
      dispatch({
        type: ACTIONS.CHECK_UPGRADE_SUCCESS,
        data: {
          upgradeAvailable: true, // <- here
          remoteVersion,
        },
      });
```

- Run `yarn build`
- Install `deb` package from `/lbry-desktop/dist/electron`
- Open up the recent installed version with `lbry`

## PR Checklist

<details><summary>Toggle...</summary>

What kind of change does this PR introduce?

- [x] Bugfix
- [x] Feature
- [ ] Code style update (formatting)
- [ ] Refactoring (no functional changes)
- [ ] Documentation changes
- [ ] Other - Please describe:

Please check all that apply to this PR using "x":

- [x] I have checked that this PR is not a duplicate of an existing PR (open, closed or merged)
- [ ] I added a line describing my change to CHANGELOG.md
- [x] I have checked that this PR does not introduce a breaking change
- [ ] This PR introduces breaking changes and I have provided a detailed explanation below

</details>

### CODE REVIEW

**Comment by jessopb** [comment]
Good progress. I can still get into double downloads. Flow feels good otherwise.

**Comment by Ruk33** [comment]
Thank you @jessopb for the review. I have sent a fix for it.

**Comment by jessopb** [comment]
I tested this on linux by `yarn build` this branch after changing the version in package.json to 0.52.4. Then installing that build from the .deb.
Unfortunately, it's throwing an error looking for `app-update.yml` when `lbry` is launched from console  and never shows the update available notifications. It does provide, in help page, the suggestion to download the lastest version which navigates the user to the download website.
To confirm it was this branch, I built master locally in the same way, with version set to 0.52.4 as well, and the update notifications and nags were present in that case. And there were no console complaints about `app-update.yml` 
It doesn't look to me like something in this pr could change that, but it does ???

**Comment by Ruk33** [comment]
Hello @jessopb. The error you are facing isn't coming from this PR but for this: https://github.com/lbryio/lbry-desktop/pull/7492/files#diff-9e864f33e9f6d8e9a52d5df4de9861f47a3a45222f1f8fb620acd9c37193bb12R198
Notice how we previously, for `.deb` installations, we check for updates using `Native.getAppVersionInfo().then(success, fail);` but after the change, it always uses the `ipcRenderer.send('check-for-updates', ...)` call.

**Comment by Ruk33** [comment]
To prevent having this error, you can create the following file `app-update.yml` in `/opt/LBRY/resources`

```
owner: lbryio
repo: lbry-desktop
provider: github
```

And re-open the app. With that, the error no longer shows up:

image

**Comment by jessopb** [comment]
This may be the right track, but I'm not sure how this is a solution fit for production. The problem is not only the console error, but also the lack of update notification in the app, which did not happen before, if I recall correctly.
https://github.com/lbryio/lbry-desktop/pull/7502#issuecomment-1066832556

Perhaps the solution is this sort of file copy? https://www.electron.build/configuration/contents#extraresources

**Comment by Ruk33** [comment]
@jessopb I have sent an update for the missing `app-update.yml`. After reading, it seems like just adding it as a additional resource is the best option. Will keep testing tomorrow to make sure everything is ok.

**Comment by jessopb** [comment]
I'm seeing windows misreporting the title of the prerelease as a release.
I'm seeing deb misreporting the release instead of prerelease in the "begin upgrade" modal `sudo... ` 
Also, closing the upgrade modal (reexposing the nag) and opening it again reinitiates the download.

File: ui/component/lastReleaseChanges/view.jsx [hunk 1]
```diff
@@ -35,7 +36,7 @@ const LastReleaseChanges = (props: Props) => {
   );
 
   useEffect(() => {
-    const lastReleaseUrl = 'https://api.github.com/repos/lbryio/lbry-desktop/releases/latest';
+    const lastReleaseUrl = `https://api.github.com/repos/lbryio/lbry-desktop/releases/tags/${releaseVersion}`;
```
Comment (jessopb) [inline ui/component/lastReleaseChanges/view.jsx:39]: There's also something strange that happened - I pushed a tag with 'alpha' in it which ended up on top of the list, but there were no builds associated with it, so updates were failing. Probably some library code that just gets the last tag with a prerelease keyword in it. After I deleted the bad tag, it was fine.

File: package.json [hunk 2]
```diff
@@ -67,6 +67,7 @@
     "remove-markdown": "^0.3.0",
     "rss": "^1.2.2",
     "source-map-explorer": "^2.5.2",
+    "sudo-prompt": "^9.2.1",
```
Comment (jessopb) [inline package.json:70]: This repository is archived on github. Something to keep an eye on, or fork for our selves.

File: ui/redux/selectors/app.js [hunk 3]
```diff
@@ -6,12 +6,20 @@ export const selectState = (state) => state.app || {};
...
     case 'darwin':
       return 'https://lbry.com/get/lbry.dmg';
     case 'linux':
-      return 'https://lbry.com/get/lbry.deb';
+      // releaseVersion can be used as the tag name
+      // Example: v0.53.5-alpha.test7495b
+      // When downloading, we need to remove the initial
+      // v, ending up with a file name like
+      // LBRY_0.53.5-alpha.test7495b.deb
+      const fileName = 'LBRY_' + (releaseVersion || '').replace('v', '') + '.deb';
```
Comment (jessopb) [inline ui/redux/selectors/app.js:21]: This could maybe use some error catching/validation. It's fine though.

**Comment by jessopb** [comment]
<frodo after the thing is done>

**Pull request merged**